Errors on the number of waves?

Some errors were observed locally during the update for #25, but not on the action tests. More investigation is needed.

```
======================================= FAILURES ========================================
____________ TestPeriodicControlFunction.test_number_of_waves[100-triangle] _____________

self = <test_control_pulseoptim.TestPeriodicControlFunction object at 0x7f7808e7d590>
pulse_type = 'TRIANGLE', n_waves = 100

    @pytest.mark.parametrize("n_waves", [1, 5, 10, 100])
    def test_number_of_waves(self, pulse_type, n_waves):
        kwargs = {
            "num_tslots": self.num_tslots,
            "evo_time": self.evo_time,
            "init_pulse_type": pulse_type,
            "init_pulse_params": {"num_waves": n_waves},
            "gen_stats": False,
        }
        system = _merge_kwargs(hadamard, kwargs)
>       assert _count_waves(system) == n_waves
E       AssertionError: assert np.int64(98) == 100
E        +  where np.int64(98) = _count_waves(_System(system=Quantum object: dims=[[2], [2]], shape=(2, 2), type='oper', dtype=CSR, isherm=True\nQobj data =\n[[ 1.  0...se, 'init_pulse_type': 'TRIANGLE', 'fid_err_targ': 1e-10, 'dyn_type': 'UNIT', 'init_pulse_params': {'num_waves': 100}}))

tests/test_control_pulseoptim.py:626: AssertionError
________________ TestPeriodicControlFunction.test_frequency[10-triangle] ________________

self = <test_control_pulseoptim.TestPeriodicControlFunction object at 0x7f7808e89810>
pulse_type = 'TRIANGLE', frequency = 10

    @pytest.mark.parametrize("frequency", [0.1, 1, 10, 20])
    def test_frequency(self, pulse_type, frequency):
        kwargs = {
            "num_tslots": self.num_tslots,
            "evo_time": self.evo_time,
            "init_pulse_type": pulse_type,
            "init_pulse_params": {"freq": frequency},
            "fid_err_targ": 1e-5,
            "gen_stats": False,
        }
        system = _merge_kwargs(hadamard, kwargs)
>       assert _count_waves(system) == self.evo_time * frequency
E       AssertionError: assert np.int64(98) == (10 * 10)
E        +  where np.int64(98) = _count_waves(_System(system=Quantum object: dims=[[2], [2]], shape=(2, 2), type='oper', dtype=CSR, isherm=True\nQobj data =\n[[ 1.  0...': False, 'init_pulse_type': 'TRIANGLE', 'fid_err_targ': 1e-05, 'dyn_type': 'UNIT', 'init_pulse_params': {'freq': 10}}))
E        +  and   10 = <test_control_pulseoptim.TestPeriodicControlFunction object at 0x7f7808e89810>.evo_time

tests/test_control_pulseoptim.py:639: AssertionError
________________ TestPeriodicControlFunction.test_frequency[20-triangle] ________________

self = <test_control_pulseoptim.TestPeriodicControlFunction object at 0x7f7808e8a510>
pulse_type = 'TRIANGLE', frequency = 20

    @pytest.mark.parametrize("frequency", [0.1, 1, 10, 20])
    def test_frequency(self, pulse_type, frequency):
        kwargs = {
            "num_tslots": self.num_tslots,
            "evo_time": self.evo_time,
            "init_pulse_type": pulse_type,
            "init_pulse_params": {"freq": frequency},
            "fid_err_targ": 1e-5,
            "gen_stats": False,
        }
        system = _merge_kwargs(hadamard, kwargs)
>       assert _count_waves(system) == self.evo_time * frequency
E       AssertionError: assert np.int64(198) == (10 * 20)
E        +  where np.int64(198) = _count_waves(_System(system=Quantum object: dims=[[2], [2]], shape=(2, 2), type='oper', dtype=CSR, isherm=True\nQobj data =\n[[ 1.  0...': False, 'init_pulse_type': 'TRIANGLE', 'fid_err_targ': 1e-05, 'dyn_type': 'UNIT', 'init_pulse_params': {'freq': 20}}))
E        +  and   10 = <test_control_pulseoptim.TestPeriodicControlFunction object at 0x7f7808e8a510>.evo_time

tests/test_control_pulseoptim.py:639: AssertionError
__________________ TestPeriodicControlFunction.test_frequency[20-saw] ___________________

self = <test_control_pulseoptim.TestPeriodicControlFunction object at 0x7f7808e8a850>
pulse_type = 'SAW', frequency = 20

    @pytest.mark.parametrize("frequency", [0.1, 1, 10, 20])
    def test_frequency(self, pulse_type, frequency):
        kwargs = {
            "num_tslots": self.num_tslots,
            "evo_time": self.evo_time,
            "init_pulse_type": pulse_type,
            "init_pulse_params": {"freq": frequency},
            "fid_err_targ": 1e-5,
            "gen_stats": False,
        }
        system = _merge_kwargs(hadamard, kwargs)
>       assert _count_waves(system) == self.evo_time * frequency
E       AssertionError: assert np.int64(199) == (10 * 20)
E        +  where np.int64(199) = _count_waves(_System(system=Quantum object: dims=[[2], [2]], shape=(2, 2), type='oper', dtype=CSR, isherm=True\nQobj data =\n[[ 1.  0...stats': False, 'init_pulse_type': 'SAW', 'fid_err_targ': 1e-05, 'dyn_type': 'UNIT', 'init_pulse_params': {'freq': 20}}))
E        +  and   10 = <test_control_pulseoptim.TestPeriodicControlFunction object at 0x7f7808e8a850>.evo_time

tests/test_control_pulseoptim.py:639: AssertionError
=================================== warnings summary ====================================
../../../local/bli/conda/test/lib/python3.11/site-packages/qutip/__init__.py:24
  /local/bli/conda/test/lib/python3.11/site-packages/qutip/__init__.py:24: UserWarning: matplotlib not found: Graphics will not work.
    warnings.warn("matplotlib not found: Graphics will not work.")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```
```
QuTiP Version:      5.1.0.dev0+418e338
Numpy Version:      2.0.0
Scipy Version:      1.13.1
Cython Version:     None
Matplotlib Version: 3.9.0
Python Version:     3.12.3
Number of CPUs:     40
BLAS Info:          Generic
INTEL MKL Ext:      False
Platform Info:      Linux (x86_64)
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -36,14 +36,12 @@ jobs:
     - name: Install QuTiP from PyPI
       if: ${{ matrix.qutip-version != '' && ! startsWith( matrix.qutip-version, '@') }}
       run: |
-        python -m pip install 'numpy>=1.19,<2.0'
         python -m pip install ${{ matrix.qutip-pre }} 'qutip${{ matrix.qutip-version }}'
 
     - name: Install QuTiP from GitHub
       if: ${{ startsWith( matrix.qutip-version, '@') }}
       run: |
         python -m pip install scipy cython
-        python -m pip install 'numpy>=1.19,<2.0'
         python -m pip install 'git+https://github.com/qutip/qutip.git${{ matrix.qutip-version }}'
 
     - name: Install qutip-qtrl

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -23,7 +23,7 @@ jobs:
             python-version: '3.10'
           - case-name: qutip-5.0
             os: ubuntu-latest
-            qutip-version: '==5.0.1'
+            qutip-version: '==5.0.*'
             python-version: '3.9'
 
     steps:

--- a/tests/test_control_pulseoptim.py
+++ b/tests/test_control_pulseoptim.py
@@ -610,7 +610,7 @@ def _count_waves(system):
     ],
 )
 class TestPeriodicControlFunction:
-    num_tslots = 1_000
+    num_tslots = 3_000
     evo_time = 10
 
     @pytest.mark.parametrize("n_waves", [1, 5, 10, 100])

